### PR TITLE
fix: handle thrift versions where headers may be unset

### PR DESCRIPTION
**Pull request by hsheth2**

### CODE REVIEW

File: pyhive/hive.py [hunk 1]
```diff
@@ -118,7 +118,12 @@ def _parse_timestamp(value):
 # This was taken and modified from https://github.com/dropbox/PyHive/pull/325/files#r412841634.
 class TCookieHttpClient(thrift.transport.THttpClient.THttpClient):
     def flush(self):
-        cookies = self.headers.get_all('Set-Cookie')
+        if not hasattr(self, 'headers'):
+            # Compatibility with older versions of thrift.
+            # See https://github.com/apache/thrift/commit/103a11c9c28ac963a3b2591ecac641db3cbaa113.
```
Comment (jinlintt) [inline pyhive/hive.py:123]: I found referencing this commit a bit confusing.
If we want to say this is for compatible with older thrift, i.e., `<= 0.16.0`, then shouldn't we reference the `0.16.0` code where `self.headers` is only initialized after the first request has been sent and therefore could be undefined when the code below is called.
https://github.com/apache/thrift/blob/2a93df80f27739ccabb5b885cb12a8dc7595ecdf/lib/py/src/transport/THttpClient.py#L187

File: pyhive/hive.py [hunk 2]
```diff
@@ -118,7 +118,12 @@ def _parse_timestamp(value):
...
+            self.headers = None
+
+        cookies = self.headers.get_all('Set-Cookie') if self.headers else None
```
Comment (jinlintt) [inline pyhive/hive.py:126]: Maybe `if hasattr(self, 'headers') and self.headers` will be cleaner and easier to read.

**Comment by hsheth2** [comment]
@jinlintt Agreed on both comments. Pushed up some changes that make this more clear

**Review by jinlintt — APPROVED** [review]
Looks great! Thank you for the quick fix!

**Pull request merged**